python/sglang/bench_offline_throughput.py

```diff
@@ -0,0 +1,288 @@
+            type=int,
+            default=BenchArgs.sharegpt_output_len,
+            help="Output length for each request. Overrides the output length from the ShareGPT dataset.",
+        )
+        parser.add_argument(
+            "--random-input-len",
+            type=int,
+            default=BenchArgs.random_input_len,
+            help="Number of input tokens per request, used only for random dataset.",
+        )
+        parser.add_argument(
+            "--random-output-len",
+            type=int,
+            default=BenchArgs.random_output_len,
+            help="Number of output tokens per request, used only for random dataset.",
+        )
+        parser.add_argument(
+            "--random-range-ratio",
+            type=float,
+            default=BenchArgs.random_range_ratio,
+            help="Range of sampled ratio of input/output length, "
+            "used only for random dataset.",
+        )
+        parser.add_argument(
+            "--gen-num-groups",
+            type=int,
+            default=BenchArgs.gen_num_groups,
+            help="Number of groups with shared prefix, used"
+            "only for generate-shared-prefix",
+        )
+        parser.add_argument(
+            "--gen-prompts-per-group",
+            type=int,
+            default=BenchArgs.gen_prompts_per_group,
+            help="Number of prompts per group of shared prefix, used"
+            "only for generate-shared-prefix",
+        )
+        parser.add_argument(
+            "--gen-system-prompt-len",
+            type=int,
+            default=BenchArgs.gen_system_prompt_len,
+            help="System prompt length, used" "only for generate-shared-prefix",
+        )
+        parser.add_argument(
+            "--gen-question-len",
+            type=int,
+            default=BenchArgs.gen_question_len,
+            help="Question length, used" "only for generate-shared-prefix",
+        )
+        parser.add_argument(
+            "--disable-ignore-eos",
+            type=bool,
+            default=BenchArgs.disable_ignore_eos,
+            help="Disable ignore EOS token",
+        )
+        parser.add_argument("--seed", type=int, default=1, help="The random seed.")
+
+    @classmethod
+    def from_cli_args(cls, args: argparse.Namespace):
+        # use the default value's type to case the args into correct types.
+        attrs = [(attr.name, type(attr.default)) for attr in dataclasses.fields(cls)]
+        print(attrs)
+        return cls(
+            **{attr: attr_type(getattr(args, attr)) for attr, attr_type in attrs}
+        )
+
+
+def throughput_test_once(
+    backend_name: str,
+    backend: Union[Engine, Runtime],
+    reqs: List[Tuple[str, int, int]],
+    output_len: int,
+    ignore_eos: bool,
+):
+    measurement_results = {
+        "total_input_tokens": sum(r[1] for r in reqs),
+    }
+
+    prompt = [r[0] for r in reqs]
+    sampling_params = {
+        "temperature": 0,
+        "max_new_tokens": output_len,
```

Comment:
The output lens are different for different requests.

Reply:
is there any way to pass in different output lens for the same `.generate` call?
```
backend.generate(prompt=prompt, sampling_params=sampling_params)
```

Reply:
It is supported in the underlying APIs. Can you update the function signature of Engine:generate?
https://github.com/sgl-project/sglang/blob/218ab3611ddf46ce6acf8a465611a01faa275eb7/python/sglang/srt/managers/io_struct.py#L39-L40